src/runtime/button-api.js

```diff
@@ -116,6 +125,14 @@ export class ButtonApi {
     }
     button.setAttribute('title', msg(TITLE_LANG_MAP, button) || '');
     button.addEventListener('click', callback);
+    if (this.deps_.config().analyticsMode == AnalyticsMode.IMPRESSIONS) {
+      this.deps_.entitlementsManager().getEntitlements().then(entitlements => {
+        // isFromUserAction = False
+        this.analyticsService_.setReadyToPay(entitlements.isReadyToPay);
+        this.analyticsService_.logEvent(
+            AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON);
```

Comment:
you can send the event without getting entitlements and ready to pay. 
The analytics Service will be set with ready to pay bit already if entitlements fetch was done.

Is there a reason why you need the ready to pay?

Reply:
I'd just do if (this.deps_.config().analyticsMode == AnalyticsMode.IMPRESSIONS) {
                    this.analyticsService_.logEvent(AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON);
}

Reply:
isReadyToPay being logged was a request by Shams. He mentioned that we needed to log this information, and we should couple it in this launch.

Reply:
Leaving the entitlements promise check but removing the setReadyToPay call for now.

Reply:
the pitfall of waiting for an entitlements fetch is
1. You will send the SUBSCRIBE BUTTON IMPRESSION when entitlements response comes back, not when the button is rendered
2. Ideally, you want to be honest about the data you are collecting. Meaning; If the publisher fetched entitlements, and the entitlements response was available at the time the subscribe button was rendered, then you log that information.  But if entitlements fetch was not done or the response was not available, when the subscribe button was rendered, your data should say so. 
For these two reasons, you should remove the entitlements check too.

Reply:
Done. Thanks for the explanation!